src/api/Metadata.js

```diff
@@ -352,6 +353,96 @@ class Metadata extends File {
         return editors;
     }
 
+    /**
+     * Utility to concat instance and template into one entity.
+     *
+     * @param {Object} instance - metadata instance
+     * @param {Object} template - metadata template
+     * @return {Object} metadata template instance
+     */
+    createTemplateInstance(instance: MetadataInstanceV2, template: MetadataTemplate): MetadataTemplateInstance {
+        const metadataFields: MetadataInstanceTemplateField[] = [];
+
+        if (template.templateKey !== METADATA_TEMPLATE_PROPERTIES) {
+            // Get Metadata Fields for Instances created from predefinied template
+            const templateFields = template.fields || [];
+            templateFields.map(async field => {
+                metadataFields.push({
+                    description: field.description,
+                    displayName: field.displayName,
+                    hidden: field.hidden || field.isHidden,
+                    id: field.id,
+                    key: field.key,
+                    options: field.options,
+                    type: field.type,
+                    value: instance[field.key],
+                });
+            });
+        } else {
+            // Get Metadata Fields for Custom Instances
+            Object.keys(instance).forEach(key => {
+                if (!key.startsWith('$')) {
+                    // $FlowFixMe
+                    metadataFields.push({
+                        key,
+                        type: 'string',
+                        value: instance[key],
+                    });
+                }
+            });
+        }
+
+        return {
+            displayName: template.displayName,
+            hidden: template.hidden || template.isHidden,
+            id: template.id,
+            metadataFields,
+            scope: template.scope,
+            templateKey: template.templateKey,
+        };
+    }
+
+    /**
+     * Creates and returns metadata entities.
+     *
+     * @param {string} id - Box file id
+     * @param {Array} instances - metadata instances
+     * @param {Object} customPropertiesTemplate - custom properties template
+     * @param {Array} enterpriseTemplates - enterprise templates
+     * @param {Array} globalTemplates - global templates
+     * @return {Array} metadata editors
+     */
+    async getTemplateInstances(
+        id: string,
+        instances: Array<MetadataInstanceV2>,
+        customPropertiesTemplate: MetadataTemplate,
+        enterpriseTemplates: Array<MetadataTemplate>,
+        globalTemplates: Array<MetadataTemplate>,
+    ): Promise<Array<MetadataTemplateInstance>> {
+        // Get all usable templates for metadata instances
+        const templates: Array<MetadataTemplate> = [customPropertiesTemplate].concat(
+            enterpriseTemplates,
+            globalTemplates,
+        );
+
+        // Filter out classification
+        const filteredInstances = this.extractClassification(id, instances);
```

Comment:
Would it make more sense to only pass filteredInstances to this function instead? Seems like that should be the callers responsibility.

Reply:
I've moved filtering to `getMetadata` and now I am passing `filteredInstances` to `getEditors` and `getTemplateInstances` as `instances` prop.